src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionProcessingServiceImpl.java

```diff
@@ -91,7 +91,7 @@ public void processActionRequest(final Action action) throws CTPException {
 
       ActionRequest actionRequest = prepareActionRequest(action);
 
-      if (actionRequest != null) {
+      if (actionRequest != null & validator.validate(actionRequest)) {
```

Comment:
Is it true that we don't want to process any action requests for completed/completed by phone cases?

Reply:
Also the log in the else for this if statement isn't entirely accurate now as there's now new ways to hit it

Reply:
the else is related to being an unfound action type rather than the message being invalid.

Reply:
if the case is completed/completedbyphone we don't want a reminder/letter, which is what these action requests trigger.

Reply:
:+1: 
The else will also be hit if the action request is completed/completed by phone now aswell though right?

Reply:
if (valid(actionType)) {
      ActionDTO.ActionEvent event = actionType.getResponseRequired() ?
          ActionDTO.ActionEvent.REQUEST_DISTRIBUTED : ActionDTO.ActionEvent.REQUEST_COMPLETED;

      transitionAction(action, event);

      ActionRequest actionRequest = prepareActionRequest(action);

      if (actionRequest != null & validator.validate(actionRequest)) {
        actionInstructionPublisher.sendActionInstruction(actionType.getHandler(), actionRequest);
      }

      // advise casesvc to create a corresponding caseevent for our action
      caseSvcClientService.createNewCaseEvent(action, CategoryDTO.CategoryName.ACTION_CREATED);
    } else {
      log.error("Unexpected situation. actionType is not defined for action with actionid {}", action.getId());
    }

Reply:
Ah cool I was getting confused but the way github was laying it out 👍

Reply:
Yeh the wrapping makes it awkward.